python_modules/libraries/dagster-aws/dagster_aws/pipes/message_readers.py

```diff
@@ -216,6 +217,15 @@ def no_messages_debug_text(self) -> str:
         )
 
 
+def get_log_events(client: "CloudWatchLogsClient", **log_params):
```

Comment:
I've mad `max_retries` configurable. I've left it in a separate function because it's used in two places, `tail_cloudwatch_events` and `PipesCloudWatchMessageReader.download_messages()`. Happy to inline it in both places if you'd prefer though?

Reply:
Oh I also increased the default max retries to 10, as noticed when it was 5 I would ocassionally still get the ThrottlingException on the output.

Reply:
I wonder if we should sleep between requests? `get_log_events` returns batches so it should be fine right? How about we sleep for `0.5s` (can be configurable)?

Reply:
I wonder if we should sleep between requests? `get_log_events` returns batches so it should be fine right? How about we sleep for `0.5s` (can be configurable)?

Reply:
I wonder if we should sleep between requests? `get_log_events` returns batches so it should be fine right? How about we sleep for `0.5s` (can be configurable)?

Reply:
The quota in our AWS account for 'GetLogEvents throttle limit in transactions per second' is 10 per second, I guess that's the standard across accounts as AWS mentions that it's not eligible for a quota increase. I believe, that's across the account as well so probably don't want to use up all 10 within a single second. So yes, I think a sleep between requests makes a lot of sense. 

Just to clarify, are you suggesting a sleep() just before we call the existing backoff() or we could provide a custom `delay_generator` which just sleeps between retries, or something else completely?